server/src/data_store.cpp

```diff
@@ -159,7 +179,7 @@ bool DataStore::GetPosition(const std::string& id, QueryT query_by, Position& po
         else
         {
             console_->error("Failed to read from database");
-            exit(1);
+            return false;
```

Comment:
I haven't looked into `sqlite3_finalize`. Is there a problem if it is not called when `sqlite3_prepare` has been called?